iree/tools/CMakeLists.txt

```diff
@@ -121,6 +121,18 @@ iree_cc_test(
 
 if(${IREE_BUILD_COMPILER})
 
+  # Logic to identify targets_to_link adopted from MLIR:
+  # Manually expand the target library, since our MLIR libraries
+  # aren't plugged into the LLVM dependency tracking. If we don't
+  # do this then we can't insert the CodeGen library after ourselves
+  llvm_expand_pseudo_components(TARGET_LIBS AllTargetsCodeGens)
+  # Prepend LLVM in front of every target, this is how the library
+  # are named with CMake
+  set(targets_to_link)
+  foreach(t ${TARGET_LIBS})
+    list(APPEND targets_to_link "LLVM${t}")
+  endforeach(t)
```

Comment:
I really dislike this change, since it pulls in `LLVMWebAssemblyCodeGen` and I think it is the reason for blowing up the size of `iree-opt` (and probably other targets depending on `iree_opt_library`, haven't checked). At the moment we set
https://github.com/google/iree/blob/18acd53ee4f4348bfd5f336835a9855dbbce8972/build_tools/cmake/iree_copts.cmake#L131

I have no clew so far if we want to change `LLVM_TARGETS_TO_BUILD` or if there is any better solution.